source/particle/generator/quadrature_points.cc

```diff
@@ -0,0 +1,130 @@
+/*
+  Copyright (C) 2015 by the authors of the ASPECT code.
+
+ This file is part of ASPECT.
+
+ ASPECT is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2, or (at your option)
+ any later version.
+
+ ASPECT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with ASPECT; see the file doc/COPYING.  If not see
+ <http://www.gnu.org/licenses/>.
+ */
+
+#include <aspect/particle/generator/quadrature_points.h>
+
+#include <deal.II/fe/fe_values.h>
+
+namespace aspect
+{
+  namespace Particle
+  {
+    namespace Generator
+    {
+      template <int dim>
+      QuadraturePoints<dim>::QuadraturePoints()
+      {
+        starting_particle_index = 0;
+      }
+
+      template <int dim>
+      void
+      QuadraturePoints<dim>::generate_particles(std::multimap<types::LevelInd, Particle<dim> > &particles)
+      {
+        const QGauss<dim> quadrature_formula(this->get_parameters().stokes_velocity_degree + 1);
+
+        unsigned int n_particles_to_generate = 0, prefix_sum = 0;
```

Comment:
Also please change the type to `types::particle_index`, see below for a justification